python/lsst/ts/atpneumaticssimulator/pneumatics_simulator.py

```diff
@@ -239,6 +387,179 @@ async def write_noack_response(self, sequence_id: int) -> None:
+
+        items["id"] = tel_id
+        await self.telemetry_server.write_json(data={**items})
+
+    async def cmd_evt_connect_callback(self, server: tcpip.OneClientServer) -> None:
+        """Callback function for when a cmd/evt client connects or disconnects.
+
+        When a cmd/evt client connects, background tasks are started and events
+        are sent.
+        When the cmd/evt client disconnects, all background tasks get stopped.
+        """
+        if server.connected:
+            await self.configure()
+            await self.initialize()
+
+    async def tel_connect_callback(self, server: tcpip.OneClientServer) -> None:
+        """Callback function for when a tel client connects or disconnects.
+
+        When a tel client connects, the telemetry loop is started.
+        When the tel client disconnects, the telemetry loop is stopped.
+        """
+        if server.connected:
+            if self._telemetry_task.done():
+                self._telemetry_task = asyncio.create_task(self.telemetry_loop())
+        else:
+            self._telemetry_task.cancel()
+
+    async def set_cell_vents_events(self, closed: bool, opened: bool) -> None:
+        """Set m1VentsLimitSwitches, m1VentsPosition and cellVentsState events.
+
+        Output any changes.
+
+        Parameters
+        ----------
+        closed : `bool`
+            Are the closed switches active?
+        opened : `bool`
+            Are the opened switches active?
+        """
+        assert not (closed and opened)
+        if not (closed or opened):
+            self.cell_vents_state = ATPneumatics.CellVentState.INMOTION
+            await self._write_evt(
+                evt_id=Event.CELLVENTSTATE, state=self.cell_vents_state
+            )
+
+        self.m1_vents_limit_switches.ventsClosedActive = closed
+        self.m1_vents_limit_switches.ventsOpenedActive = opened
+        await self._write_evt(
+            evt_id=Event.M1VENTSLIMITSWITCHES,
+            ventsClosedActive=self.m1_vents_limit_switches.ventsClosedActive,
+            ventsOpenedActive=self.m1_vents_limit_switches.ventsOpenedActive,
+        )
+        if opened:
+            self.m1_vents_position = ATPneumatics.VentsPosition.OPENED
+            self.cell_vents_state = ATPneumatics.CellVentState.OPENED
+            await self._write_evt(
+                evt_id=Event.CELLVENTSTATE, state=self.cell_vents_state
+            )
+        elif closed:
+            self.m1_vents_position = ATPneumatics.VentsPosition.CLOSED
+            self.cell_vents_state = ATPneumatics.CellVentState.CLOSED
+            await self._write_evt(
+                evt_id=Event.CELLVENTSTATE, state=self.cell_vents_state
+            )
+        else:
+            self.m1_vents_position = ATPneumatics.VentsPosition.PARTIALLYOPENED
+        await self._write_evt(
+            evt_id=Event.M1VENTSPOSITION, position=self.m1_vents_position
+        )
```

Comment:
If possible, please set cell_vents_state here, instead of above in a separate `if not (closed or open)` statement.

If your current order is necessary (as it may well be) then please add a code comment explaining why, so future maintainers won't mess it up.